config/initializers/lograge.rb

```diff
@@ -5,7 +5,7 @@
   config.lograge.formatter = Lograge::Formatters::Json.new
   config.lograge.custom_options = lambda do |event|
     {
-      request_id: event.payload[:headers]['action_dispatch.request_id']
+      request_id: event.payload[:headers]&.[]('action_dispatch.request_id')
```

Comment:
As this wasn't explained in code to describe this elegant approach to check that a key exists in `payload`, leaving this breadcrumb in the PR for those who may look at this line in the future and in case it isn't clear what this is.

The `&.` is known as [Safe Navigation Operator](https://ruby-doc.org/core-3.0.2/doc/syntax/calling_methods_rdoc.html#label-Safe+Navigation+Operator) and combined with [] it is equivalent of using `hash && hash[:key]` which allows safely navigating the `payload` which may not always contain the `headers` key.